Doc/library/typing.rst

```diff
@@ -1389,11 +1389,61 @@ These can be used as types in annotations. They all support subscription using
 
    Special typing construct for marking user-defined type guard functions.
 
-   ``TypeGuard`` can be used to annotate the return type of a user-defined
-   type guard function.  ``TypeGuard`` only accepts a single type argument.
-   At runtime, functions marked this way should return a boolean.
+   Type guard functions are user-defined functions that return whether their
+   argument is an instance of a particular type.
```

Comment:
I think we could potentially keep using the term "type guard functions" here, if it's clearly not generic but specific to `TypeGuard` -- but I think using a different generic term here would be better here as well.

Reply:
Switching to "type predicate" here too for consistency.